### ISSUE TITLE
Alternate functions assignable to unused pins

### ISSUE THREAD

**Issue by xygax** [RobTillaart/rotaryDecoder#10]
Rob. 
when using a single encoder and button Is it possible to reuse the unused pins as outputs. Can this be configured or available as an option see the attached for idea ( only the relevant section)

thanks
steve

[PCF8574.docx](https://github.com/RobTillaart/rotaryDecoder/files/13849648/PCF8574.docx)

**Comment by RobTillaart** [comment]
Hi Steve,
Yes using the free pins is possible, there are a number of things that needs some thinking before it can be in the library.

First the user is now free to select which set of the pins is connected to the rotaryEncoder when he chooses to connect only one.
That means there are 4 possible configurations to define the six free pins. 

In your case you want to define 3 as input and 3 as output if I understand correctly.

maybe the easiest way to solve this is to use the PCF8574 library side by side with the rotaryDecoder class. 
The PCF8574 class you use to control the pins, input, output, HIGH, LOW etc. 
You might need to mask pins. 

Alternative would be to add two functions to the class to be able to use unused pins.
- **void write1(pin, HIGH, LOW)**
- **uint8_t read1(pin)**

As I am rather busy I will come back to this issue later next week.

**Comment by RobTillaart** [comment]
Note to myself:  rotaryDecoderSwitch library should have similar additions.

**Comment by xygax** [comment]
Thanks Rob. I have hardware to do first so wont get to firmware until next month. 

In the specific example i have it is as follows  :
- B0, B1  quadrature with pull up
- B2 third input with pull up and multiplexed Led 
- B3, B4, B5 Leds output only but could have switches
- B6 multiplex direction output only
- B7 buzzer output only

This leaves the quadrature with exclusive pins 
the next 4 are potentially bidirectional and run with a shadow register for both output and input. 
B6 is the selection for direction and bit 7 is static output 

Thanks Steve
Sent from Outlook for Android

(reformatted the text to make it easy readable)

**Comment by RobTillaart** [comment]
@Steve
Just saw your message. I already created a develop branch as another task took less time. 
That allowed me to prototype this one.

The read1(pin) function should work without a problem as it doesn't alter anything.

The write1() function writes a bitmask back and that could cause problems and that need to be tested carefully.
It might interfere with the RE pins e.g. keeping pins LOW or HIGH or ....
I have no hardware setup to test it myself nearby so I just leave it as a develop branch.

Furthermore I did not write an example as the usage is straightforward and an example will probably not match the users configuration. So that is left as an exercise :)
